server/spec.go

```diff
@@ -44,28 +40,10 @@ type Server interface {
 	// Boot registers the configured endpoints and starts the server under the
 	// configured address.
 	Boot()
-	// Endpoints returns the server's configured list of endpoints. These are the
-	// custom endpoints configured by the client.
-	Endpoints() []Endpoint
-	// ErrorEncoder returns the server's error encoder. This wraps the error
-	// encoder configured by the client. Clients should not implement error
-	// logging in here them self. This is done by the server itself. Clients must
-	// not implement error response writing them self. This is done by the server
-	// itself. Duplicated response writing will lead to runtime panics.
-	ErrorEncoder() kithttp.ErrorEncoder
-	Logger() micrologger.Logger
-	// RequestFuncs returns the server's configured list of request functions.
-	// These are the custom request functions configured by the client.
-	RequestFuncs() []kithttp.RequestFunc
-	// Router returns a HTTP handler for the server. The returned router will have
-	// all endpoints registered that are listed in the endpoint collection.
-	Router() *mux.Router
-	// ServiceName returns the name of the micro-service implementing the microkit
-	// server. This is used for logging and instrumentation.
-	ServiceName() string
+	// Config returns the servers configuration as given by the client.
+	Config() Config
 	// Shutdown stops the running server gracefully.
 	Shutdown()
-	TransactionResponder() microtransaction.Responder
```

Comment:
Clients will no longer have to implement all this, which is a big win for the implementation.